packages/c/cumem/xmake.lua

```diff
@@ -0,0 +1,21 @@
+package("cumem")
+set_kind("library", { headeronly = true })
+set_description("CUDA Memory Management Wrapper with Type Safety")
+
+set_homepage("https://github.com/BinhaoQin/cuMem")
+
+add_urls("https://github.com/BinhaoQin/cuMem.git")
+add_versions("1.0.0", "843500929889c08468c2af546b9ae3123b1e86c1")
+
+on_install(function(package)
+	local configs = {}
+	if package:config("shared") then
+		configs.kind = "shared"
+	end
+	import("package.tools.xmake").install(package, configs)
+end)
+
+on_test(function(package)
+	-- TODO check includes and interfaces
+	-- assert(package:has_cfuncs("foo", {includes = "foo.h"})
```

Comment:
need test

Reply:
For testing, I need to build a test target and run it. However, I'm using `clang` to build CUDA instead of standard toolchain, and I can't use `add_cugencodes("native")` to let the test be built on users' machine. Is there any way to circumvent it?

Reply:
you need `    add_deps("cuda", {system = true})`, see thrust package.

And you need to at least pass the test locally and then submit the results.

please run command to test it.
```
xmake l scripts\test -vD --shallow cumem
```